runtime/src/main/jni/ObjectManager.cpp

```diff
@@ -17,8 +17,14 @@ using namespace v8;
 using namespace std;
 using namespace tns;
 
-ObjectManager::ObjectManager(jobject javaRuntimeObject)
-    : m_javaRuntimeObject(javaRuntimeObject), m_env(JEnv()), m_numberOfGC(0), m_currentObjectId(0), m_cache(NewWeakGlobalRefCallback, DeleteWeakGlobalRefCallback, 1000, this) {
+ObjectManager::ObjectManager(jobject javaRuntimeObject) :
+        m_javaRuntimeObject(javaRuntimeObject),
+        m_env(JEnv()),
+        m_numberOfGC(0),
+        m_currentObjectId(0),
+        m_cache(NewWeakGlobalRefCallback, DeleteWeakGlobalRefCallback, 1000, this),
+        m_markingMode(JavaScriptMarkingMode::Full) {
```

Comment:
get marking mode from java static configuration, don't set it as default here, because it's easier to change defaults only in one place